NewForce-at-Mountwest/bangazon-api-theapocalypseclams
# Allow developers to access a product resource

# Description

Verbs to be supported

GET
POST
PUT
DELETE
User should be able to GET a list of products and GET a single product.

Testing Criteria
Write a testing class and test methods that validate the GET single, GET all, POST, PUT, and DELETE operations work as expected.

Steps to Test

MANUAL TEST SETUP FOR A DATABASE WITH EXISTING DATA:
Setup the database
Step 1:
In Visual Studio create a new query on the BanagzonAPI database.

Step 2:
Copy, paste, and run the following SQL script to populate the database.
SQL Script

Built in tests
Step 1:
In Visual Studio type test in the search bar and click on "Test Explorer"

Step 2:
Open the dropdown menu on TestBangazonAPI

Step 3:
Locate and select Product

Step 4:
Click on the single arrow (run) / control+ R, T which with Product selected will only run the tests in view.

Step 5:
Verify that all customer tests ran and pass (green checks)

MANUAL TEST:

GET Manual Test:
After setting up the test data above, you should be able to do a CTRL F5 from VisualStudio and change the URL to api/product and see the 2 records you just added above - this would be the manual test for GET.

GetProduct Manual TEST:
To Manually test for GetProduct - do the same as above and add /1 or /2 to the end of the URL after pressing CTRL-F5.
you should see just one record in the browser.

POST Manual TEST:
To Manually test the POST process Use POSTMAN again and remove the /2 from the end of the URL, change your command to POST and in the BODY add the following and click SEND:
{
        "ProductTypeId" = 1,
                "CustomerId" = 2,
                "Price" = 1.99m,
                "Title" = "Bacon Lettuce Tomato Sandwhich",
                "Description" = "Test Product Description for Edit",
                "Quantity" = 2
}

Put Manual TEST:
To Manually test PUT, open POSTMAN and add the same URL from the Browser above (https://localhost:44352/api/product/2), change your command to PUT and in the body type:
{
        "ProductTypeId" = 2,
                "CustomerId" = 2,
                "Price" = 1.99m,
                "Title" = "Test Product Edit",
                "Description" = "Test Product Description for Edit",
                "Quantity" = 1
}
Then press SEND.

DELETE_ONE Manual TEST:
To Manually Test the DELETE process, use POSTMAN again, change your Command to DELETE and add a /2 to the end of the URL and click SEND.

Checklist:
[ x] I have commented my code, particularly in hard-to-understand areas
[ x] I have made corresponding changes to the documentation
[ x] My changes generate no new errors
[ x] I have added tests that prove my fix is effective or that my feature works
[ x] New and existing unit tests pass locally with my changes
